### PR TITLE
ENH: Add variable cell support to julip.py Python interface

### PR DESCRIPTION
**Pull request by jameskermode**
This PR updates the ASE interface to include variable cell minimisation.

### CODE REVIEW

File: temp/julip.py [hunk 1]
```diff
@@ -18,12 +18,13 @@
...
 ASECalculator = julia.eval('ASECalculator(c) = JuLIP.ASE.ASECalculator(c)')
 fixedcell = julia.eval('fixedcell(a) = JuLIP.Constraints.FixedCell(a)')
+variablecell = julia.eval('variablecell(a) = JuLIP.Constraints.VariableCell(a)')
 
 class JulipCalculator(Calculator):
     """
     ASE-compatible Calculator that calls JuLIP.jl for forces and energy
     """
-    implemented_properties = ['forces', 'energy']
+    implemented_properties = ['forces', 'energy', 'stress']
```
Comment (cortner) [inline temp/julip.py]: should we try to make this depend on what the Julia calculator has actually implemented?

**Comment by cortner** [comment]
can we just have a quick chat tomorrow to make sure stresses in ASE and JuLIP are the same thing (I never checked myself)

**Comment by jameskermode** [comment]
Yup sure. Stress in ASE has units of eV/A^3, so it is given by sigma_ij = 1/det R_ij  \partial E / \partial R_ij, where R_ij is the cell matrix (with lattice vectors as rows).

File: temp/julip.py (hunk 1 above)
Comment (jameskermode) [inline temp/julip.py]: Good idea, but this is a class attribute, which has to be constant. I think the best we can do is throw a NotImplementedError exception if the required quantity is not implemented.

**Comment by cortner** [comment]
So just to confirm: If I call `F = R'` i.e. `F` has lattice vectors as columns, then for me  
* [virial]_ij = - \partial E / \partial F_ij
* stress = - virial / det(F) 
* stress_ase = stress'
correct?

**Comment by jameskermode** [comment]
Yes - but stress is always symmetric so last transpose is not needed (actually stored as 6-element vector in ASE).

**Comment by jameskermode** [comment]
Non-symmetric contributions to strain and stress correspond to rotation, but no deformation.

**Comment by cortner** [comment]
so I've rewritten JuLIP stress and virial, but haven't yet merged it. Will it be straightforward to merge your PR, then rebase my branch, and then fix the ASE interface?

**Comment by jameskermode** [comment]
Yup, or you can merge yours first and then I'll update this PR before merging it.

**Comment by cortner** [comment]
whichever will be easier - maybe with your suggestion you don't have to teach me how to rebase :)

**Comment by jameskermode** [comment]
yup, go ahead, and let me know when to merge into this PR branch from master.

**Comment by cortner** [comment]
done.

**Comment by cortner** [comment]
Here is the stress in bulk Si with the new SW calculator
screen shot 2016-10-12 at 07 40 36

**Comment by jameskermode** [comment]
The stress matches QUIP's SW numerically, I'll do a couple more tests then we can merge this PR.

**Comment by cortner** [comment]
is this ready to be merged?

**Comment by jameskermode** [comment]
Tests with SW energy volume curve vs. QUIP look good. Ready to merge.

**Comment by jameskermode** [comment]
Currently rather slower than QUIP SW (before #24):
h j4rrk8mxbcqaaaabjru5erkjggg

**Comment by cortner** [comment]
Ok at some point when we have nothing better to do we should fix this!

**Pull request merged**